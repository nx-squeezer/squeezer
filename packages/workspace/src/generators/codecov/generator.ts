import { formatFiles, Tree } from '@nrwl/devkit';
import fetch from 'node-fetch-commonjs';

import { existsGitHubCiWorkflow, addGitHubCiJobStep } from '../github-workflow';
import { addBadgeToReadme, addImplicitDependencyToNxConfig, getGitRepoSlug } from '../lib';
import { writeProjectsToCodecov, getCodecovFile, readRawCodecov } from './codecov';

export async function codecovGenerator(tree: Tree) {
  writeProjectsToCodecov(tree);
  addImplicitDependencyToNxConfig(tree, { [getCodecovFile(tree)]: '*' });

  if (existsGitHubCiWorkflow(tree)) {
    addGitHubCiJobStep(tree, 'test', {
      name: 'Codecov',
      uses: 'codecov/codecov-action@v3.1.0',
      if: `hashFiles('coverage/**/*') != ''`,
      with: {
        fail_ci_if_error: true,
        verbose: true,
      },
    });
  } else {
    console.error(`Codecov needs to be called from a CI pipeline, but it could not be found.`);
    console.error(`Try to generate it first using nx g @nx-squeezer/workspace:github-workflow`);
  }

  // Validate
  const response = await fetch(`https://api.codecov.io/validate`, { method: 'POST', body: readRawCodecov(tree) });
  if (response.ok) {
    console.log(`Codecov file was validated successfully`);
  } else {
    throw new Error(`Couldn't generate a valid Codecov file`);
  }

  console.log(`You can configure the Codecov app at: https://github.com/apps/codecov`);

  const gitRepoSlug = getGitRepoSlug(tree);
  if (gitRepoSlug == null) {
    console.error(`Could not add badge to README, remote repo could not be detected.`);
  } else {
    addBadgeToReadme(
      tree,
      `https://codecov.io/gh/${gitRepoSlug}/branch/main/graph/badge.svg`,
      `https://codecov.io/gh/${gitRepoSlug}`,
      'codecov'
    );
  }

  await formatFiles(tree);
}
