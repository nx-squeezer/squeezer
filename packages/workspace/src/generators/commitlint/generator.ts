import { formatFiles, installPackagesTask, readJson, Tree, writeJson } from '@nrwl/devkit';

import {
  commitlintCli,
  CommitlintConfig,
  commitlintConfigConventional,
  commitlintConfigPath,
  commitlintDefaultConfig,
} from './commitlint';
import { addHuskyToPackageJson, addDevDependencyToPackageJson, installHuskyTask, addHuskyHookTask } from '../lib';

export async function commitlintGenerator(tree: Tree) {
  addHuskyToPackageJson(tree);
  addDevDependencyToPackageJson(tree, commitlintCli);
  addDevDependencyToPackageJson(tree, commitlintConfigConventional);

  const existingConfiguration: Partial<CommitlintConfig> = tree.exists(commitlintConfigPath)
    ? readJson(tree, commitlintConfigPath)
    : {};

  writeJson(tree, commitlintConfigPath, { ...existingConfiguration, ...commitlintDefaultConfig });

  await formatFiles(tree);

  return () => {
    installPackagesTask(tree);
    installHuskyTask(tree);
    addHuskyHookTask(tree, 'commit-msg', 'npx --no-install commitlint --edit $1');
  };
}
