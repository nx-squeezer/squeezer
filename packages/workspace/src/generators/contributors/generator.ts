import { formatFiles, Tree } from '@nx/devkit';

import { readmeFile, getGitRepoSlug } from '@nx-squeezer/devkit';

export async function contributorsGenerator(tree: Tree) {
  if (!tree.exists(readmeFile)) {
    throw new Error(`Missing Readme file at: ${readmeFile}`);
  }

  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  let readme = tree.read(readmeFile)!.toString();

  const repoSlug = getGitRepoSlug(tree);

  if (repoSlug == null) {
    throw new Error(`Remote repo could not be detected.`);
  }

  if (readme.includes(`## Contributors`)) {
    console.log(`Contributors section already existing at: ${readmeFile}`);
    return;
  }

  const link = `https://contrib.rocks/image?repo=${repoSlug}`;
  const image = `https://github.com/${repoSlug}/graphs/contributors`;
  readme += `## Contributors\n\n[![contributors](${link})](${image})\n`;

  tree.write(readmeFile, readme);

  await formatFiles(tree);
}
