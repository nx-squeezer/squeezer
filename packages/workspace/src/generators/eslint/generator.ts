import { formatFiles, installPackagesTask, Tree } from '@nrwl/devkit';
import { JSONSchemaForESLintConfigurationFiles } from '@schemastore/eslintrc';
import {
  addDevDependencyToPackageJson,
  addEsLintPlugin,
  addEsLintRules,
  lintWorkspaceTask,
  readEsLintConfig,
  writeEsLintConfig,
} from '../core';
import { EsLintGeneratorSchema } from './schema';

export default async function (tree: Tree, options: EsLintGeneratorSchema) {
  let eslintConfig: JSONSchemaForESLintConfigurationFiles = readEsLintConfig(tree);

  if (options.eslintRecommended) {
    eslintConfig = addEsLintRecommendedRules(eslintConfig);
  }

  if (options.sonarJs) {
    addDevDependencyToPackageJson(tree, 'eslint-plugin-sonarjs');
    eslintConfig = addEsLintPlugin(eslintConfig, 'sonarjs');
    eslintConfig = addSonarJsRecommendedRules(eslintConfig);
  }

  writeEsLintConfig(tree, eslintConfig);
  await formatFiles(tree);

  return () => {
    installPackagesTask(tree);
    lintWorkspaceTask(tree);
  };
}

function addEsLintRecommendedRules(
  eslintConfig: JSONSchemaForESLintConfigurationFiles
): JSONSchemaForESLintConfigurationFiles {
  let newEslintConfig: JSONSchemaForESLintConfigurationFiles = { ...eslintConfig };

  const env: Exclude<JSONSchemaForESLintConfigurationFiles['env'], undefined> = { ...(eslintConfig.env, {}) };
  env.node = true;
  env.browser = true;
  env.es2022 = true;

  newEslintConfig = { env, ...newEslintConfig };

  newEslintConfig = addEsLintRules(newEslintConfig, {
    files: ['*.ts', '*.tsx', '*.js', '*.jsx'],
    extends: ['eslint:recommended'],
    rules: {},
  });

  return newEslintConfig;
}

function addSonarJsRecommendedRules(
  eslintConfig: JSONSchemaForESLintConfigurationFiles
): JSONSchemaForESLintConfigurationFiles {
  let newEslintConfig: JSONSchemaForESLintConfigurationFiles = { ...eslintConfig };

  newEslintConfig = addEsLintRules(newEslintConfig, {
    files: ['*.ts', '*.tsx', '*.js', '*.jsx'],
    extends: ['plugin:sonarjs/recommended'],
    rules: {},
  });

  return newEslintConfig;
}
