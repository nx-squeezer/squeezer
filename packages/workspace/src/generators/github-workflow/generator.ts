import { formatFiles, generateFiles, names, Tree } from '@nx/devkit';

import {
  joinNormalize,
  addImplicitDependencyToNxConfig,
  addScriptToPackageJson,
  getGitRepo,
  addBadgeToReadme,
  ciFile,
} from '@nx-squeezer/devkit';

import { GitHubWorkflowGeneratorSchema } from './schema';

export async function gitHubWorkflowGenerator(tree: Tree, options: GitHubWorkflowGeneratorSchema) {
  if (!options.force && tree.exists(ciFile)) {
    console.log(`GitHub workflow already existing at path: ${ciFile}`);
    return;
  }

  if (options.useNxCloud) {
    console.log(`In order to use Nx Cloud add a secret with the NX_CLOUD_AUTH_TOKEN`);
    console.log(`It is suggested that you create a new token for this purpose with read/write access.`);
    console.log(`You can configure it at: https://cloud.nx.app/`);
    console.log(`Also consider using Nx Cloud GitHub app to access run results: https://github.com/apps/nx-cloud`);
    console.log();
  }

  const targets = ['build', 'lint', 'test', 'e2e'].map((target) => {
    const formattedNames = names(target);
    formattedNames.className = formattedNames.className.replace(/e2e/i, 'e2e');
    return formattedNames;
  });
  const templateOptions = { ...options, targets, tmpl: '' };

  generateFiles(tree, joinNormalize(__dirname, 'files'), '.github/workflows', templateOptions);
  addImplicitDependencyToNxConfig(tree, { '.github/workflows/*.yml': '*' });
  addScriptToPackageJson(tree, 'nx', 'nx');
  addScriptToPackageJson(tree, 'lint:workspace', 'nx workspace-lint');

  const gitRepo = getGitRepo(tree);
  if (gitRepo == null) {
    console.error(`Could not add badge to README, remote repo could not be detected.`);
  } else {
    addBadgeToReadme(
      tree,
      `${gitRepo}/actions/workflows/ci.yml/badge.svg`,
      `${gitRepo}/actions/workflows/ci.yml`,
      'CI',
      true
    );
  }

  await formatFiles(tree);
}
