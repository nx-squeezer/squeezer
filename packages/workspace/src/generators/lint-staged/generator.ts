import { formatFiles, installPackagesTask, Tree, readJson, writeJson } from '@nx/devkit';

import {
  addHuskyToPackageJson,
  addDevDependencyToPackageJson,
  installHuskyTask,
  addHuskyHookTask,
} from '@nx-squeezer/devkit';

import { lintStaged, LintStagedConfig, lintStagedConfigPath, lintStagedDefaultConfig } from './lint-staged';

export async function lintStagedGenerator(tree: Tree) {
  addHuskyToPackageJson(tree);
  addDevDependencyToPackageJson(tree, lintStaged);

  const existingConfiguration: LintStagedConfig = tree.exists(lintStagedConfigPath)
    ? readJson(tree, lintStagedConfigPath)
    : {};

  writeJson(tree, lintStagedConfigPath, { ...existingConfiguration, ...lintStagedDefaultConfig });

  await formatFiles(tree);

  return () => {
    installPackagesTask(tree);
    installHuskyTask(tree);
    addHuskyHookTask(tree, 'pre-commit', 'npx lint-staged');
  };
}
