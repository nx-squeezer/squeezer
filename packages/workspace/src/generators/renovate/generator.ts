import { formatFiles, generateFiles, Tree } from '@nrwl/devkit';
import { parseDocument, Scalar, stringify, YAMLSeq } from 'yaml';

import { ciFile } from '../github-workflow';
import { getGitRepoSlug, joinNormalize } from '../lib';
import { renovateCiFile, renovateBranch } from './renovate';
import { renovateConfigValidatorTask } from './renovate-config-validator-task';
import { RenovateGeneratorSchema } from './schema';

export async function renovateGenerator(tree: Tree, options: RenovateGeneratorSchema) {
  if (!options.force && tree.exists(renovateCiFile)) {
    console.log(`Renovate workflow already existing at path: ${renovateCiFile}`);
    return;
  }

  if (options.useNxCloud) {
    console.log(`In order to use Nx Cloud add a secret with the NX_CLOUD_AUTH_TOKEN`);
    console.log(`It is suggested that you create a new token for this purpose with read/write access.`);
    console.log(`You can configure it at: https://cloud.nx.app/`);
    console.log(`Also consider using Nx Cloud GitHub app to access run results: https://github.com/apps/nx-cloud`);
    console.log();
  }

  const gitRepoSlug = getGitRepoSlug(tree);
  if (gitRepoSlug == null) {
    throw new Error(`Could not identify GitHub repo slug.`);
  }

  // Generate files

  const templateOptions = { ...options, gitRepoSlug, assignee: options.assignee ?? '', tmpl: '' };
  generateFiles(tree, joinNormalize(__dirname, 'files'), '.', templateOptions);

  if (options.local) {
    generateFiles(tree, joinNormalize(__dirname, 'presets'), '.', templateOptions);
  }

  // Update CI

  if (!tree.exists(ciFile)) {
    throw new Error(`Renovate needs a GitHub workflow CI file, none found at: ${ciFile}`);
  }

  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  const ci = parseDocument(tree.read(ciFile)!.toString());
  const pushBranches: YAMLSeq<Scalar> = ci.getIn(['on', 'push', 'branches']) as YAMLSeq<Scalar>;
  if (!pushBranches.items.map((item) => item.value).includes(renovateBranch)) {
    pushBranches.add(new Scalar(renovateBranch));
  }
  tree.write(ciFile, stringify(ci));

  await formatFiles(tree);

  return () => {
    renovateConfigValidatorTask(tree);
  };
}
