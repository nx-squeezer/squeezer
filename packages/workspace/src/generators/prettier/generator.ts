import { formatFiles, installPackagesTask, readJson, Tree, writeJson } from '@nx/devkit';
import { SchemaForPrettierrc } from '@schemastore/prettierrc';

import { addDevDependencyToPackageJson, lintWorkspaceTask, formatWorkspaceTask } from '@nx-squeezer/devkit';

import { prettierPlugin, eslintPluginPrettier, prettierConfigJsonFile, prettierConfigFile } from './prettier';
import { prettierDefaultConfig } from './prettier-default-config';
import { addEsLintPlugin, addEsLintRules, isEsLintPluginPresent } from '../eslint';

export default async function prettierGenerator(tree: Tree) {
  setPrettierConfig(tree);

  if (isEsLintPluginPresent(tree, prettierPlugin)) {
    return;
  }
  addEsLintPlugin(tree, prettierPlugin, '@nx');
  addEsLintRules(tree, {
    files: ['*.ts', '*.tsx', '*.js', '*.jsx', '*.json', '*.md', '*.html'],
    extends: ['plugin:prettier/recommended'],
    rules: {},
  });

  addDevDependencyToPackageJson(tree, eslintPluginPrettier);
  await formatFiles(tree);

  return () => {
    installPackagesTask(tree);
    lintWorkspaceTask(tree);
    formatWorkspaceTask(tree);
  };
}

function setPrettierConfig(tree: Tree): void {
  if (tree.exists(prettierConfigJsonFile)) {
    return;
  }

  if (!tree.exists(prettierConfigFile)) {
    writeJson(tree, prettierConfigFile, {});
  }

  let prettierConfig = readJson<Exclude<SchemaForPrettierrc, string>>(tree, prettierConfigFile);
  prettierConfig = { ...prettierDefaultConfig, ...prettierConfig };

  writeJson(tree, prettierConfigFile, prettierConfig);
  tree.rename(prettierConfigFile, prettierConfigJsonFile);
}
