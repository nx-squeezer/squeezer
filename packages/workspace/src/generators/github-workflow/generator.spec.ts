import { NxJsonConfiguration, readJson, Tree } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { parse } from 'yaml';

import {
  getGitRepo,
  nxConfigFile,
  readmeFile,
  readPackageJson,
  existsGitHubCiWorkflow,
  ciFile,
} from '@nx-squeezer/devkit';

import { gitHubWorkflowGenerator } from './generator';
import { gitHubWorkflowSchematic } from './generator.compat';

jest.mock('@nx-squeezer/devkit', () => ({
  ...jest.requireActual('@nx-squeezer/devkit'),
  getGitRepo: jest.fn(),
}));

describe('@nx-squeezer/workspace github workflow generator', () => {
  let tree: Tree;

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace();
    jest.spyOn(console, 'log').mockImplementation(() => null);
    jest.spyOn(console, 'error').mockImplementation(() => null);
    (getGitRepo as jest.Mock).mockReturnValue('https://github.com/test/test');
  });

  it('should run successfully', async () => {
    await gitHubWorkflowGenerator(tree, { branch: 'main', useNxCloud: true, force: true });

    expect(existsGitHubCiWorkflow(tree)).toBeTruthy();
  });

  it('should provide a schematic', async () => {
    expect(typeof gitHubWorkflowSchematic({ branch: 'main', useNxCloud: true, force: true })).toBe('function');
  });

  it('should skip execution if a GitHub CI workflow already exists', async () => {
    tree.write(ciFile, '');

    await gitHubWorkflowGenerator(tree, { branch: 'main', useNxCloud: true, force: false });

    expect(console.log).toHaveBeenCalledWith(`GitHub workflow already existing at path: ${ciFile}`);
  });

  it('should not skip execution if a GitHub CI workflow already exists but passing force option', async () => {
    tree.write(ciFile, '');

    await gitHubWorkflowGenerator(tree, { branch: 'main', useNxCloud: true, force: true });

    expect(tree.read(ciFile)?.toString()).toBeTruthy();
  });

  it('should generate the GitHub CI workflow', async () => {
    await gitHubWorkflowGenerator(tree, { branch: 'main', useNxCloud: true, force: true });

    const ci = parse(tree.read(ciFile)?.toString() ?? '');

    expect(ci).toStrictEqual(ciFileContent());
  });

  it('should declare the nx script in package.json', async () => {
    await gitHubWorkflowGenerator(tree, { branch: 'main', useNxCloud: true, force: true });

    const packageJson = readPackageJson(tree);

    expect(packageJson.scripts?.nx).toBe('nx');
  });

  it('should declare the implicit dependency in nx.json', async () => {
    await gitHubWorkflowGenerator(tree, { branch: 'main', useNxCloud: true, force: true });

    const nxConfig = readJson<NxJsonConfiguration>(tree, nxConfigFile);

    // eslint-disable-next-line @delagen/deprecation/deprecation -- https://github.com/nx-squeezer/squeezer/issues/680
    expect(nxConfig.implicitDependencies?.['.github/workflows/*.yml']).toBe('*');
  });

  it('should add a badge in readme', async () => {
    tree.write(readmeFile, '# Readme\n');
    await gitHubWorkflowGenerator(tree, { branch: 'main', useNxCloud: true, force: true });

    const readme = tree.read(readmeFile)?.toString() ?? '';

    expect(readme).toContain('[![CI]');
  });

  it('should not add a badge in readme if repo can not be resolved', async () => {
    (getGitRepo as jest.Mock).mockReturnValue(null);
    tree.write(readmeFile, '# Readme\n');
    await gitHubWorkflowGenerator(tree, { branch: 'main', useNxCloud: true, force: true });

    const readme = tree.read(readmeFile)?.toString() ?? '';

    expect(readme).not.toContain('[![CI]');
    expect(console.error).toHaveBeenCalledWith(`Could not add badge to README, remote repo could not be detected.`);
  });
});

/* eslint-disable sonarjs/no-duplicate-string */
const ciFileContent = () => ({
  name: 'CI',
  on: {
    push: {
      branches: ['main'],
    },
    pull_request: {
      branches: ['main'],
    },
  },
  jobs: {
    npm: {
      name: 'NPM',
      'runs-on': 'ubuntu-latest',
      env: {
        NX_CLOUD_AUTH_TOKEN: '${{ secrets.NX_CLOUD_AUTH_TOKEN }}',
      },
      steps: [
        {
          name: 'Checkout repo',
          uses: 'actions/checkout@v3.0.2',
        },
        {
          name: 'Setup Node.js',
          uses: 'actions/setup-node@v3.4.1',
          with: {
            'node-version': '18.x',
          },
        },
        {
          name: 'NPM',
          uses: 'ng-easy/npm-setup@v2.0.10',
        },
      ],
    },
    build: {
      name: 'Build',
      needs: ['npm'],
      'runs-on': 'ubuntu-latest',
      env: {
        NX_CLOUD_AUTH_TOKEN: '${{ secrets.NX_CLOUD_AUTH_TOKEN }}',
      },
      steps: [
        {
          name: 'Checkout repo',
          uses: 'actions/checkout@v3.0.2',
          with: {
            'fetch-depth': 0,
          },
        },
        {
          name: 'Setup Node.js',
          uses: 'actions/setup-node@v3.4.1',
          with: {
            'node-version': '18.x',
          },
        },
        {
          name: 'NPM',
          uses: 'ng-easy/npm-setup@v2.0.10',
          with: {
            'nx-key': 'build',
          },
        },
        {
          name: 'Nx Set SHAs',
          uses: 'nrwl/nx-set-shas@v3.0.0',
        },
        {
          name: 'Build',
          run: 'npx nx affected --target=build --parallel=3',
        },
      ],
    },
    lint: {
      name: 'Lint',
      needs: ['npm'],
      'runs-on': 'ubuntu-latest',
      env: {
        NX_CLOUD_AUTH_TOKEN: '${{ secrets.NX_CLOUD_AUTH_TOKEN }}',
      },
      steps: [
        {
          name: 'Checkout repo',
          uses: 'actions/checkout@v3.0.2',
          with: {
            'fetch-depth': 0,
          },
        },
        {
          name: 'Setup Node.js',
          uses: 'actions/setup-node@v3.4.1',
          with: {
            'node-version': '18.x',
          },
        },
        {
          name: 'NPM',
          uses: 'ng-easy/npm-setup@v2.0.10',
          with: {
            'nx-key': 'lint',
          },
        },
        {
          name: 'Nx Set SHAs',
          uses: 'nrwl/nx-set-shas@v3.0.0',
        },
        {
          name: 'Lint',
          run: 'npx nx affected --target=lint --parallel=3',
        },
        {
          name: 'Lint workspace',
          run: 'npm run lint:workspace',
        },
      ],
    },
    test: {
      name: 'Test',
      needs: ['npm'],
      'runs-on': 'ubuntu-latest',
      env: {
        NX_CLOUD_AUTH_TOKEN: '${{ secrets.NX_CLOUD_AUTH_TOKEN }}',
      },
      steps: [
        {
          name: 'Checkout repo',
          uses: 'actions/checkout@v3.0.2',
          with: {
            'fetch-depth': 0,
          },
        },
        {
          name: 'Setup Node.js',
          uses: 'actions/setup-node@v3.4.1',
          with: {
            'node-version': '18.x',
          },
        },
        {
          name: 'NPM',
          uses: 'ng-easy/npm-setup@v2.0.10',
          with: {
            'nx-key': 'test',
          },
        },
        {
          name: 'Nx Set SHAs',
          uses: 'nrwl/nx-set-shas@v3.0.0',
        },
        {
          name: 'Test',
          run: 'npx nx affected --target=test --parallel=3',
        },
      ],
    },
    e2e: {
      name: 'e2e',
      needs: ['npm'],
      'runs-on': 'ubuntu-latest',
      env: {
        NX_CLOUD_AUTH_TOKEN: '${{ secrets.NX_CLOUD_AUTH_TOKEN }}',
      },
      steps: [
        {
          name: 'Checkout repo',
          uses: 'actions/checkout@v3.0.2',
          with: {
            'fetch-depth': 0,
          },
        },
        {
          name: 'Setup Node.js',
          uses: 'actions/setup-node@v3.4.1',
          with: {
            'node-version': '18.x',
          },
        },
        {
          name: 'NPM',
          uses: 'ng-easy/npm-setup@v2.0.10',
          with: {
            'nx-key': 'e2e',
          },
        },
        {
          name: 'Nx Set SHAs',
          uses: 'nrwl/nx-set-shas@v3.0.0',
        },
        {
          name: 'e2e',
          run: 'npx nx affected --target=e2e --parallel=1',
        },
      ],
    },
  },
});
