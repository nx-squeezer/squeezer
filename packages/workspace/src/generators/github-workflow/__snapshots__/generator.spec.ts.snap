// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@nx-squeezer/workspace github workflow generator should generate the GitHub CI workflow 1`] = `
{
  "jobs": {
    "build": {
      "env": {
        "NX_CLOUD_AUTH_TOKEN": "\${{ secrets.NX_CLOUD_AUTH_TOKEN }}",
      },
      "name": "Build",
      "needs": [
        "npm",
      ],
      "runs-on": "ubuntu-latest",
      "steps": [
        {
          "name": "Checkout repo",
          "uses": "actions/checkout@v3.0.2",
          "with": {
            "fetch-depth": 0,
          },
        },
        {
          "name": "Setup Node.js",
          "uses": "actions/setup-node@v3.4.1",
          "with": {
            "node-version": "18.x",
          },
        },
        {
          "name": "NPM",
          "uses": "ng-easy/npm-setup@v2.0.10",
          "with": {
            "nx-key": "build",
          },
        },
        {
          "name": "Nx Set SHAs",
          "uses": "nrwl/nx-set-shas@v3.0.0",
        },
        {
          "name": "Build",
          "run": "npx nx affected --target=build --parallel=3",
        },
      ],
    },
    "e2e": {
      "env": {
        "NX_CLOUD_AUTH_TOKEN": "\${{ secrets.NX_CLOUD_AUTH_TOKEN }}",
      },
      "name": "e2e",
      "needs": [
        "npm",
      ],
      "runs-on": "ubuntu-latest",
      "steps": [
        {
          "name": "Checkout repo",
          "uses": "actions/checkout@v3.0.2",
          "with": {
            "fetch-depth": 0,
          },
        },
        {
          "name": "Setup Node.js",
          "uses": "actions/setup-node@v3.4.1",
          "with": {
            "node-version": "18.x",
          },
        },
        {
          "name": "NPM",
          "uses": "ng-easy/npm-setup@v2.0.10",
          "with": {
            "nx-key": "e2e",
          },
        },
        {
          "name": "Nx Set SHAs",
          "uses": "nrwl/nx-set-shas@v3.0.0",
        },
        {
          "name": "e2e",
          "run": "npx nx affected --target=e2e --parallel=1",
        },
      ],
    },
    "lint": {
      "env": {
        "NX_CLOUD_AUTH_TOKEN": "\${{ secrets.NX_CLOUD_AUTH_TOKEN }}",
      },
      "name": "Lint",
      "needs": [
        "npm",
      ],
      "runs-on": "ubuntu-latest",
      "steps": [
        {
          "name": "Checkout repo",
          "uses": "actions/checkout@v3.0.2",
          "with": {
            "fetch-depth": 0,
          },
        },
        {
          "name": "Setup Node.js",
          "uses": "actions/setup-node@v3.4.1",
          "with": {
            "node-version": "18.x",
          },
        },
        {
          "name": "NPM",
          "uses": "ng-easy/npm-setup@v2.0.10",
          "with": {
            "nx-key": "lint",
          },
        },
        {
          "name": "Nx Set SHAs",
          "uses": "nrwl/nx-set-shas@v3.0.0",
        },
        {
          "name": "Lint",
          "run": "npx nx affected --target=lint --parallel=3",
        },
      ],
    },
    "npm": {
      "env": {
        "NX_CLOUD_AUTH_TOKEN": "\${{ secrets.NX_CLOUD_AUTH_TOKEN }}",
      },
      "name": "NPM",
      "runs-on": "ubuntu-latest",
      "steps": [
        {
          "name": "Checkout repo",
          "uses": "actions/checkout@v3.0.2",
        },
        {
          "name": "Setup Node.js",
          "uses": "actions/setup-node@v3.4.1",
          "with": {
            "node-version": "18.x",
          },
        },
        {
          "name": "NPM",
          "uses": "ng-easy/npm-setup@v2.0.10",
        },
      ],
    },
    "test": {
      "env": {
        "NX_CLOUD_AUTH_TOKEN": "\${{ secrets.NX_CLOUD_AUTH_TOKEN }}",
      },
      "name": "Test",
      "needs": [
        "npm",
      ],
      "runs-on": "ubuntu-latest",
      "steps": [
        {
          "name": "Checkout repo",
          "uses": "actions/checkout@v3.0.2",
          "with": {
            "fetch-depth": 0,
          },
        },
        {
          "name": "Setup Node.js",
          "uses": "actions/setup-node@v3.4.1",
          "with": {
            "node-version": "18.x",
          },
        },
        {
          "name": "NPM",
          "uses": "ng-easy/npm-setup@v2.0.10",
          "with": {
            "nx-key": "test",
          },
        },
        {
          "name": "Nx Set SHAs",
          "uses": "nrwl/nx-set-shas@v3.0.0",
        },
        {
          "name": "Test",
          "run": "npx nx affected --target=test --parallel=3",
        },
      ],
    },
  },
  "name": "CI",
  "on": {
    "pull_request": {
      "branches": [
        "main",
      ],
    },
    "push": {
      "branches": [
        "main",
      ],
    },
  },
}
`;
